independent-projects/arc/runtime/src/main/java/io/quarkus/arc/BuiltInBean.java

```diff
@@ -0,0 +1,45 @@
+package io.quarkus.arc;
+
+import java.lang.annotation.Annotation;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.context.spi.CreationalContext;
+
+/**
+ * Common class for all built-in beans.
+ *
+ */
+public abstract class BuiltInBean<T> implements InjectableBean<T> {
+
+    @Override
+    public String getIdentifier() {
+        return sha1(toString());
```

Comment:
I don't think we need such a complicated id for built-in beans. I would use some prefix and bean class name instead. Something like `builtin_BeanManagerBean`...